### DISCUSSION TITLE
Why the download:plugins takes so long to complete !

### DISCUSSION THREAD

**Discussion by lmcbout** [eclipse-theia/theia#7662] [General] [answered]
During a compilation, Why does it take 50-70% of the time to download plugins and sometimes , it takes ~30 sec for the same download.
Statistic for the download:plugins takes ~ 8-10 minutes to complete: There is no warning of anythings display on the screen , you have to let it go and wait 
```
Done in 390.93s.
Done in 511.92s.
```
I added the following plugins in package.json and this is where it fails, 

        "vscode-java-debug": "https://github.com/microsoft/vscode-java-debug/releases/download/0.24.0/vscjava.vscode-java-debug-0.24.0.vsix",
        "vscode-java-test": "https://github.com/microsoft/vscode-java-test/releases/download/0.22.0/vscjava.vscode-java-test-0.22.0.vsix",
        "vscode-java-dependency-viewer": "https://github.com/microsoft/vscode-java-dependency/releases/download/0.6.0/vscode-java-dependency-0.6.0.vsix",
        "vscode-java-redhat": "https://open-vsx.org/api/redhat/java/0.59.1/file/redhat.java-0.59.1.vsix",
        "vscode-coverage-gutters": "https://github.com/ryanluker/vscode-coverage-gutters/releases/download/v2.4.3/vscode-coverage-gutters-2.4.3.vsix",
        "vscode-maven": "https://open-vsx.org/api/vscjava/vscode-maven/0.21.0/file/vscjava.vscode-maven-0.21.0.vsix"

REMOVE THE PLUGINS FOLDER AND perform the following command: 
```
    time yarn download:plugins
```
Is it because the new plugins comes from another path:
```
   https://github.com/theia-ide/vscode-builtin- ... or 
   "https://open-vsx.org/api/...  or
   "https://github.com/microsoft/...
```
Is it because the redhat java version is not the latest on the open-vsx registry ?

**Comment by lmcbout** [comment]
Did more test and I discovered that the following plugins takes most of the download time. Without this plugins, it takes 3-15 secondes to download and when I add the other plugins, the minimal time goes up to 3 minutes or more
plugins: 
    "vscode-java-redhat": "https://open-vsx.org/api/redhat/java/0.59.1/file/redhat.java-0.59.1.vsix",

**Comment by benoitf** [comment]
@lmcbout what's your internet bandwidth ? vscode-java is 50M+ download but on my side it's taking less than 2s to grab it.

**Comment by lmcbout** [comment]
My connection bandwidth is 50 Mbps
 PING ms
49
 DOWNLOAD Mbps
59.42
 UPLOAD Mbps
49.78

**Comment by lmcbout** [comment]
@benoitf   What is strange is : If I select the "Download" button link in  the page (https://open-vsx.org/extension/redhat/java) , it takes less than 5 secondes, but when the same link is in "package.json" with the other builtin plugins, it takes an average 3-8 minutes

**Comment by benoitf** [comment]
some http_proxy / proxy in command line vs browser configuration ?

**Comment by lmcbout** [comment]
I have no proxy-setting

**Comment by vince-fugnitto** [comment]
@lmcbout it looks to be a regression in the latest update to the `download:plugins` script due to `requestretry` buffers.

**Comment by lmcbout** [comment, marked as answer]
After the merge from @marechal-p and @vince-fugnitto 
https://github.com/eclipse-theia/theia/pull/7677
This is resolved
Thanks everyone
